tools/src/main/java/org/apache/kafka/trogdor/workload/ConsumeBenchWorker.java

```diff
@@ -86,48 +89,91 @@ public void start(Platform platform, WorkerStatusTracker status,
         @Override
         public void run() {
             try {
-                HashSet<TopicPartition> partitions = new HashSet<>();
+                Map<String, List<TopicPartition>> partitionsByTopic = new HashMap<>();
                 for (Map.Entry<String, PartitionsSpec> entry : spec.activeTopics().materialize().entrySet()) {
+                    List<TopicPartition> partitions = new ArrayList<>();
+                    String topic = entry.getKey();
                     for (Integer partitionNumber : entry.getValue().partitionNumbers()) {
-                        partitions.add(new TopicPartition(entry.getKey(), partitionNumber));
+                        partitions.add(new TopicPartition(topic, partitionNumber));
                     }
+                    partitionsByTopic.put(topic, partitions);
                 }
-                log.info("Will consume from {}", partitions);
-                executor.submit(new ConsumeMessages(partitions));
+
+                AbstractConsumeMessages consumeMessagesTask;
+                if (spec.consumerGroup() == null) {
+                    spec.consumerGroup(DEFAULT_CONSUMER_GROUP);
```

Comment:
We should use a randomly generated (and hopefully unique!) consumer group here so that we don't conflict with other people running a test.